### ISSUE TITLE
Eclair node sends an `Error` instead of local force-closing a channel in data-loss-recovery protocol

### ISSUE THREAD

**Issue by btcontract** [ACINQ/eclair#727]
A user has restored an old channel state after losing a phone and initiated a data-loss-recovery procedure, but when asked to force-close a channel ACINQ node replies with `Error` which says "please publish your local commitment", so looks like it has initiated a data-loss-recivery in return.

https://github.com/btcontract/lnwallet/issues/95#issuecomment-426910342

Just in case: BLW has started to send a `ChannelReestablish` with zeroed `yourLastPerCommitmentSecret` field instead of an `Error` because LND no longer force-closes it's channels because of remote errors: https://github.com/lightningnetwork/lnd/issues/1904, perhaps this has something to do with it.

**Comment by btcontract** [comment]
Channel funding tx is `6ab8e24ef1545bd3950268436501be94055b3eea59b567896a95dee4a35ad22b`

**Comment by btcontract** [comment]
I've found out the reason: BLW was sending `Long.MaxValue` as `nextLocalCommitmentNumber` and Eclair channel was ending up in this state: https://github.com/ACINQ/eclair/blob/master/eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala#L1348-L1356

I've changed `nextLocalCommitmentNumber` to zero now but https://github.com/btcontract/lnwallet/issues/95 is stuck already... any ideas what to do about it?

Also, I don't quite understand the reasoning in provided piece of code, why not just publish a local commit since clearly something strange is going on and we are not at risk. That's what LND does currently.

**Comment by pm47** [comment]
```
counterparty says that they have a more recent commitment than the one we know of!!! 
ourCommitmentNumber=83 theirCommitmentNumber=9223372036854775807
```
😆

> I've changed nextLocalCommitmentNumber to zero now but btcontract/lnwallet#95 is stuck already... any ideas what to do about it?

How about I force close the channel? This is what BLW is expecting right?

> Also, I don't quite understand the reasoning in provided piece of code, why not just publish a local commit since clearly something strange is going on and we are not at risk. That's what LND does currently.

Well I think the rationale is explained pretty clearly in the comment: maybe the remote is saying the truth and we are actually behind? In that case a local close will be considered a cheating attempt and we lose everything. Or maybe the counterparty is wrong like in this example (dishonest or buggy), and then we want a local commit, but it should be the least frequent case and requires careful review.

**Comment by btcontract** [comment]
> How about I force close the channel? This is what BLW is expecting right?

Yes, but it may not be that simple given that eclair channel is in that specific state which prevents local force-closing.

**Comment by pm47** [comment]
Done, txid=`4d1628c6a0668144b1eb7f22780a55bea92949c86898a3be86160a1969965904`
